### ISSUE TITLE
APPROACH OM being included in the report when it was not suppose to be

### ISSUE THREAD

**Issue by torv72** [torv72/torv-reports-v4#4]
Per your instructions, I ran the Sonnenalp Golf Club in the main branch.  I had to update the MASTER DATABASE prior to running the report. 

Unfortunately, it did the same thing.  It is recognizing OM APPROACH even though I never took any samples like that this time.  Everything appears to be correct in the MASTER DATABASE.  

I ran another Sonnenalp Golf Club report with the date 2023-10-18.  It also recognized OM APPROACH even though I did not take any samples.  

So I believe this might be an existing error in the code that needs to be further investigated.  I just never caught it for some reason.

This is the code I used to create the report:

```
generate_report(.site_name = "Sonnenalp Golf Club",
                .zip_code = 81632,
                .date_sample_submitted = "2024-04-30",
                .start_date = "2007-01-01",
                .warm_or_cool = "cool",
                .acid_extract = "Mehlich",
                .include_results_interpretation = FALSE)
```

**Comment by z3tt** [comment]
We definitely need to look into this. If time allows for it, let's have a look together on Thursday?

**Comment by z3tt** [comment]
Some other issue is now popping up when running this related to the `case_when` logic applied to match months by seasons. 

Strangely, it works when run locally but fails when rendering. R is reporting that it couldn't find `sample_description_number_2` -- however, it is not relevant here at all. Need to investigate this.

**Comment by z3tt** [comment]
I have for now solved the issue with the `sample_description_number_2`. It was actually related to the `date_sample_submitted_date_2` which is not present in case there is only one sampling date in one of the types -> I've updated the if-else logic, but have to verify it with you.

---

Now back to the original question about OM being featured even though there are no measurements.

When I filter the full database for OM and the Sonnenalp Golf Club, I get a list of samples:

```
temp <- full_database %>%
    filter(
        sample_description_number_1 == "OM",
        # The older data in the MASTER DATABASE had OM data as "Soil";
        # Since May 2022 it reads in the sample type as "Physical".
        sample_type %in% c("Soil", "Physical")
    ) %>% filter(client_name == "Sonnenalp Golf Club")

temp
# A tibble: 39 × 17
#   row_number test_short_name test_full_name     source_filename date_sample_submitted client_name client_type sample_type site  sample_description_n…¹
#         <dbl> <chr>           <chr>              <chr>           <dttm>                <chr>       <chr>       <chr>       <chr> <chr>                 
#  1       7924 S196            Organic Matter % … PHYSb_36540_09… 2020-09-21 00:00:00   Sonnenalp … Golf        Physical    Sonn… OM                    
#  2       7925 S196            Organic Matter % … PHYSb_36540_09… 2020-09-21 00:00:00   Sonnenalp … Golf        Physical    Sonn… OM                    
#  3       7926 S196            Organic Matter % … PHYSb_36540_09… 2020-09-21 00:00:00   Sonnenalp … Golf        Physical    Sonn… OM                    
#  4       7927 S196            Organic Matter % … PHYSb_36540_09… 2020-09-21 00:00:00   Sonnenalp … Golf        Physical    Sonn… OM                    
#  5       7928 S196            Organic Matter % … PHYSb_36540_09… 2020-09-21 00:00:00   Sonnenalp … Golf        Physical    Sonn… OM                    
#  6       7929 S196            Organic Matter % … PHYSb_36540_09… 2020-09-21 00:00:00   Sonnenalp … Golf        Physical    Sonn… OM                    
#  7       9606 S196            Organic Matter % … PHYSb_36540_04… 2021-04-07 00:00:00   Sonnenalp … Golf        Physical    Sonn… OM                    
#  8       9607 S196            Organic Matter % … PHYSb_36540_04… 2021-04-07 00:00:00   Sonnenalp … Golf        Physical    Sonn… OM                    
#  9       9608 S196            Organic Matter % … PHYSb_36540_04… 2021-04-07 00:00:00   Sonnenalp … Golf        Physical    Sonn… OM                    
# 10       9609 S196            Organic Matter % … PHYSb_36540_04… 2021-04-07 00:00:00   Sonnenalp … Golf        Physical    Sonn… OM                    
# ℹ 29 more rows
# ℹ abbreviated name: ¹​sample_description_number_1
# ℹ 7 more variables: sample_description_number_2 <chr>, sample_description_number_3 <chr>, sample_description_number_4 <chr>,
#   analytical_company_name <chr>, lab_number <chr>, measurement_name <chr>, measurement_result <dbl>
# ℹ Use `print(n = ...)` to see more rows
```

The sample dates are also within the specified temporal range:

```
temp %>% select(date_sample_submitted) %>% unique()
# A tibble: 6 × 1
#   date_sample_submitted
#   <dttm>               
# 1 2020-09-21 00:00:00  
# 2 2021-04-07 00:00:00  
# 3 2021-11-04 00:00:00  
# 4 2022-10-19 00:00:00  
# 5 2023-10-18 00:00:00  
# 6 2024-04-30 00:00:00 
```

@torv72 If I understand you correctly, you don't expect an empty data frame here. Can you check if this information is not provided in the source database?

**Comment by torv72** [comment]
Some interesting behavior here.  I used the following generate_report:

```
generate_report(.site_name = "Sonnenalp Golf Club",
                .zip_code = 81611,
                .date_sample_submitted = "2024-04-30",
                .start_date = "2007-01-01",
                .om_seasons = "Spring",
                .warm_or_cool = "cool",
                .acid_extract = "Mehlich",
                .include_results_interpretation = FALSE,
                .draw_beeswarm = "Yes",
                .output = "html",
                .typeface = "Georama")
```

Before I pulled from Git, I reviewed the MASTER DATABASE and  confirmed everything that you show in the tibble and everything looks correct.  I than pulled on Git.  Everything worked.  However, in the Executive Summary, I'm seeing a section for APPROACH.  It should not be there because it has been since 2021 since I last took an APPROACH sample.  
<img width="724" alt="Screenshot 2024-07-02 at 1 15 54 PM" src="https://github.com/torv72/torv-reports-v4/assets/90870379/928e0740-64e8-4dce-91ef-236174e8b051">

Then I noticed that when I went to the Organic Matter part of the report, the only thing showing up are testing dates for 2024 in both Greens and Fairways.  There are no past years showing up.  They disappeared.  Also, the Table of past measurements have disappeared.  Another interesting note is that while the APPROACH comments show up in the Executive Summary (they should not as I did not test them since 2021), the APPROACH does not show up in the Organic Matter tables of graphics and they should not as I didn't take those samples.  Just an observation that is hopefully helpful.

Maybe you're just working on this and that is why they disappeared?  If you were, no worries.  Otherwise, this is what I am seeing in the html version.

<img width="1281" alt="Screenshot 2024-07-02 at 1 19 35 PM" src="https://github.com/torv72/torv-reports-v4/assets/90870379/e1e67b7b-d85c-44d0-8ac6-89459f5d7f82">
<img width="1213" alt="Screenshot 2024-07-02 at 1 20 00 PM" src="https://github.com/torv72/torv-reports-v4/assets/90870379/c3f5bce0-361d-4187-ada8-490cbc3ff38c">

**Comment by z3tt** [comment]
No, I haven't been working on this (I'd never push unfinished changes) and I was and I am seeing the same in the report. 

**For the executive summary:** 

It states that the APPROACH sampling is from 2021 so that seems okay? 
I can't find any logic for that in the script but it sounds as you only want to list the results from the current year / latest sampling date? 

I have added a comment in [`om-text-automation.R`, ll. 18](https://github.com/torv72/torv-reports-v4/blob/e55424af40fb6e6aa014d27dc4939d0bcccddebc/aux-scripts/om-text-automation.R#L18). I guess that's where we have to adjust the logic. But as I don't know the database and reference calculations in detail, I haven't changed the code yet.

**For the OM section:** 

As we've decided a while ago, the table is not featured in the web version of the report (see #20), that's okay. 
The plots are fine, too. As you have specified "Spring" as the OM season and the only two sampling dates are Oct 2023 and Apr 2024, one point per soil depth is exactly what has been reported for the spring seasons.

If you set `.om_seasons = "all"` then both sampling dates are included:
![Screenshot 2024-07-08 at 15 13 30](https://github.com/torv72/torv-reports-v4/assets/7523556/b7d479dd-a0af-4816-b411-7484dbc411fc)

**Comment by z3tt** [comment]
Hi @torv72, can you please review this issue and comment on it or close it in case it is solved / not relevant anymore? Thanks!

**Comment by torv72** [comment]
OK I have reviewed everything and have run several sample reports.  The graphics all seem to work without any issues.  I tested "Spring", "Autumn", and "all".  Everything looks good.  

With that all being said, I think I know what is happening.  There is a logic issue somewhere where Organic Matter is being pulled into the report no matter what.  It should only be included and more specifically the areas that were tested  if I actually conducted tests on the sample date specified.  Otherwise there should not be anything generated.  The water portion of the report works perfect.  If there is no water test, no summary or graphics are generated for that particular report date.

I ran a sample report for Maroon Creek Club with a sample date of 2024-06-06.  Here is the parameters:

generate_report(.site_name = "Maroon Creek Club",
                .zip_code = 81611,
                .date_sample_submitted = "2024-06-06",
                .start_date = "2017-01-01",
                .om_seasons = "Spring",
                .warm_or_cool = "cool",
                .acid_extract = "Mehlich",
                .include_results_interpretation = FALSE,
                .include_sand_fraction = TRUE,
                .draw_beeswarm = "Yes",
                .output = "pdf")


It produced the following output:

<img width="826" alt="Screenshot 2024-09-03 at 3 43 22 PM" src="https://github.com/user-attachments/assets/b7f46260-7815-4078-8cb9-0be8635a1f20">

On this particular testing date (2024-06-06), I only tested the 0-2 cm depth.  However, notice how it brought in the ROUGH, and the TEE, and the other GREEN depth measurements (2-4 cm and 4-6 cm) from the past year.

Can we take a closer look at that logic?  I don't think there is anything wrong with the MASTER DATABASE at this time.  

I suppose another option would be to always include Organic Matter in its entirety with every report and it can serve as a reminder to the client.  

If it is easier to discuss via online meeting, I can make that happen.

**Comment by z3tt** [comment]
Well spotted! Yes, we can have a look together. This might be due to the new seasonal filter which is applied in some but not all cases (?)

**Comment by torv72** [comment]
Thanks.  I get something right every once in awhile!

**Comment by z3tt** [comment]
I've looked into the problem of APPROACH being listed in the executive summary again. It turns out, that handling such a case was already implemented but the date comparison was always returning `FALSE`. I fixed the comparison and now the entry still appears but with a note that this is only shown as a reference since no sampling was conducted on the specified date:

![Screenshot 2024-09-06 at 10 14 20](https://github.com/user-attachments/assets/4af4818f-0f84-4e72-89a6-65d3dc3a2f27)

Is this still the preferred workflow or should that part be removed? I think it's fine to keep it as you are able to remove that from the .txt file in case it seems superfluous. Otherwise, you would need to search for those values in the visualizations and type it yourself. If you never need to report values for reference, we should remove it of course.
